Add endpoints for updating patches & tasks

savannaendicott opened the pull request:
### This pull requests adds coverage for the following endpoints to evergreen.py:
- POST [/tasks/{task_id}/restart](https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#restart-a-task)
- POST [/tasks/{task_id}/abort](https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#abort-a-task)
- PATCH [/tasks/{task_id}](https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#change-a-tasks-execution-status)
- POST [/patches/{patch_id}/configure](https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#configureschedule-a-patch)

### Refactoring included:
Refactored `_call_api` to be used for all requests -- instead of only GET. Another option would be to rename the method to reflect that it's only used for GET and add separate methods for POST & PATCH, however this would involve the same amount of line changes.

dbradf commented on src/evergreen/api.py:
```diff
@@ -361,6 +409,30 @@ def patches_by_project(self, project_id: str, params: Dict = None) -> Iterable[P
+
+        :param patch_id: Id of the patch to update
+        :param variants: list of objects with keys "id" who's value is the variant ID, and key "tasks"
+        with value of a list of task names to configure for specified variant. See the documentation for more details
```
Since this param string extends to multiple lines, our docs parser expects the lines after the first to be indented.

dbradf reviewed (changes requested):
Could you update the version in `pyproject.toml` and add a description of this change in `CHANGELOG.md`?

dbradf commented:
Let me know when this is ready and I can merge it for you. After it gets merged, it should auto-deploy to pypi within a few minutes.

savannaendicott commented:
> Let me know when this is ready and I can merge it for you. After it gets merged, it should auto-deploy to pypi within a few minutes.

Hi @dbradf I have made a minor additional change, I had added "variants" as an optional parameter for configuring patches but it is required by the endpoint. This is now fixed, and **ready to merge**.

dbradf commented:
evergreen merge
